Chapter5/io_handler.erl

```diff
@@ -0,0 +1,29 @@
+-module(io_handler).
+-export([init/1, terminate/1, handle_event/2]).
+
+init(Count) -> Count.
+terminate(Count) -> {count, Count}.
+
+handle_event({raise_alarm, Id, Alarm}, Count) ->
+	print(alarm, Id, Alarm, Count),
+	Count+1;
+
+handle_event({clear_alarm, Id, Alarm}, Count) ->
+	print(clear, Id, Alarm, Count),
+	Count+1;
+
+handle_event(Event, Count) -> Count.
+
+print(Type, Id, Alarm, Count) ->
+	Date = fmt(date()), Time = fmt(time()),
+	io:format("#~w,~s,~s,~w,~w,~p~n",
+	[Count, Date, Time, Type, Id, Alarm]).
+
+fmt({AInt,BInt,CInt}) ->
+	AStr = pad(integer_to_list(AInt)),
+	BStr = pad(integer_to_list(BInt)),
+	CStr = pad(integer_to_list(CInt)),
+	[AStr,$:,BStr,$:,CStr].
```

Comment:
The io_handler and log_handler modules are given in the chapter, it wasn't written by me